File: server/routes/trends.ts
import { Router } from 'express';
import { AIAgent } from '../agents/aiAgent.js';
import { getWorkspaceMetrics } from '../utils/metrics.js';

const router = Router();
const agent = new AIAgent();

router.post('/', async (req, res) => {
  try {
    const { industry, workspaceId = 'ws_123' } = req.body;
    
    if (!industry) {
      return res.status(400).json({ error: 'Industry is required' });
    }
    const metrics = getWorkspaceMetrics(workspaceId);
    const trends = await agent.generateTrends(industry, metrics);
    res.json(trends);
  } catch (error: any) {
    res.status(500).json({ error: error.message || 'Failed to generate trends' });
  }
});

router.get('/recommendations', (req, res) => {
  res.json({
    trend: "Protein Coffee",
    confidence: 92,
    recommendation: "Add to inventory."
  });
});

export default router;

File: backend/app/api/endpoints/trends.py
from fastapi import APIRouter
from app.agents.trend_agent import run_trend_analysis

router = APIRouter()

@router.get("/")
def get_trends():
    return [
        {
            "trend": "Protein Coffee",
            "confidence": 92,
            "recommendation": "Add to inventory."
        }
    ]

@router.get("/recommendations")
async def get_trend_recommendations():
    return await run_trend_analysis()
